cmd/samples/pageflow/workflow.go

```diff
@@ -0,0 +1,143 @@
+package main
+
+import (
+	"context"
+	"github.com/pborman/uuid"
+	"go.uber.org/cadence"
+	"go.uber.org/cadence/activity"
+	"go.uber.org/cadence/workflow"
+	"go.uber.org/zap"
+	"time"
+)
+
+/**
+ * This sample workflow demonstrates how a page flow can be powered by Cadence workflow.
+ */
+
+const (
+	// ApplicationName is the task list for this sample
+	ApplicationName = "pageflow"
+	SignalName = "trigger-signal"
+	QueryName = "state"
+)
+
+type State string
+
+const (
+	Initialized State = "initialized"
+	Received = "received"
+	Created = "created"
+	SubmissionReceived = "submission-received"
+	Submitted = "submitted"
+	Approved = "approved"
+	Rejected = "rejected"
+	Withdrawn = "withdrawn"
```

Comment:
is it the timeout one?

Reply:
Timeout will leave workflow at the latest status. So in practice, this need to work with workflow close status, i.e. completed/timed_out/etc.

Reply:
I see, makes sense.